src/aleph/handlers/content/vm.py

```diff
@@ -289,6 +301,50 @@ class HasParent(Protocol):
             )
 
 
+def get_extra_storage(content: InstanceContent, session: DbSession) -> int:
+    volumes: int = 0
+    for i in content.volumes:
+        try:
+            if hasattr(i, "ref") and i.ref:
+                file_pin = (
+                    session.query(FilePinDb)
+                    .filter(FilePinDb.item_hash == i.ref)
+                    .first()
+                )
+                if not file_pin:
+                    raise VmRefNotFound()
+                file_record = (
+                    session.query(StoredFileDb)
+                    .filter(StoredFileDb.hash == file_pin.file_hash)
+                    .first()
+                )
+                if file_record:
+                    volumes += file_record.size / (1024 * 1024)
+            else:
+                volumes += i.size_mib
+        except:
+            pass
```

Comment:
This should be encapsulated in a function. Several remarks:

1. Instead of using `hasattr(i, ref)`, use `isinstance(volume, ImmutableVolume`).
2. I checked the code in the repo, you can actually just look for a matching file tag by using the `get_file_tag` function. You can then directly access the file data using the `FileTag.file` property/relationship.
3. Never use `.first()` when attempting to get one result only. SQLAlchemy has `one()` and `one_or_none()`.
4. The `query()` API of SQLAlchemy is deprecated, use the `select` approach instead.
5. We store all queries in the `accessors` module, here you won't need this query but now you know for next time.
6. I don't understand why you `try .. except` this. In general it's bad practice to `except` all exceptions, we usually narrow it down. Here I guess it's because you used `first()` and tried to catch the case where there's no result. Using `one_or_none()` resolves this issue. Catching the exception is fine as well, but catch a specific exception type, not just "everything".